### DISCUSSION TITLE
Make `LockIsolated` a property wrapper?

### DISCUSSION THREAD

**Discussion by davdroman** [pointfreeco/swift-concurrency-extras#14] [Ideas]
Looking at the implementation, at first glance it seems fitting to have `LockIsolated` as a property wrapper. Is this something worth exploring?

**Comment by jshier** [comment]
Property wrappers can't completely express thread-safety, as they only wrap access to the `wrappedValue`, not to any of that value's properties. To make it safe you can add `@dynamicMemeberLookup` conformance, which accesses the `wrappedValue` through a `KeyPath`, but you have to expose such an API through the `projectedValue` and always remember to call it rather than directly access the wrapped value. This creates opportunities for unsafe usage. I recently [removed a similar feature from Alamofire's internals](https://github.com/Alamofire/Alamofire/pull/3777) because it had long been a [threat to thread-safety](https://github.com/Alamofire/Alamofire/pull/3505).
